lib/slosilo/key.rb

```diff
@@ -121,6 +121,12 @@ def stringify value
         value
       else
         value.to_json
+      end.tap do |string|
+        # Ensure that the string is encoded as raw bytes so that we don't have
+        # encoding errors when smushing it together with the salt.
+        if string.encoding != Encoding::ASCII_8BIT
+          string.force_encoding Encoding::ASCII_8BIT
+        end
```

Comment:
All text should be convertible to UTF-8; and even if that wasn't the case,
eg. if the client is using a weird (custom) encoding for tengwar or
whatever, then I'd argue an exception is expected: they would need to
encode it in binary themselves and they would know how.
9 cze 2016 01:44 "Jon Mason" <notifications@github.com> napisał(a):

> In lib/slosilo/key.rb
> <https://github.com/conjurinc/slosilo/pull/9#discussion_r66361320>:
>
> > @@ -121,6 +121,12 @@ def stringify value
> >          value
> >        else
> >          value.to_json
> > +      end.tap do |string|
> > +        # Ensure that the string is encoded as raw bytes so that we don't have
> > +        # encoding errors when smushing it together with the salt.
> > +        if string.encoding != Encoding::ASCII_8BIT
> > +          string.force_encoding Encoding::ASCII_8BIT
> > +        end
>
> Won't doing that with a string that's not compatible with UTF-8 raise an
> exception? Or are all strings convertible to UTF-8?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/conjurinc/slosilo/pull/9/files/bae37b43807606913c4d0bb77538e5d10a428152#r66361320>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAyRVMPVJEWryRLcEB5lN0QW9FZrKshBks5qJ1PHgaJpZM4IxZPf>
> .
>